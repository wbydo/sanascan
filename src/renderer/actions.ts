import * as types from "./types";

export interface Action {
  type: string;
}

export const increment: () => Action = () => {
  return {
    type: types.INCREMENT,
  };
};
