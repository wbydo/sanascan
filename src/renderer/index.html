<!DOCTYPE html>
<html lang="ja">
<head>
  <meta charset="utf-8">
  <title>YACOS</title>
  <link rel="stylesheet" href="css/style.css">
</head>
<body>
  <div id="root"></div>
</body>
</html>
