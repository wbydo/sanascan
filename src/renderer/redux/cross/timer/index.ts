import { start, kill } from "./actions";
export const actions = { start, kill };

export { middlewares } from "./middlewares";
