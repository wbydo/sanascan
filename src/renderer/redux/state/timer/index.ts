import { reducer } from "./reducers";
import * as actions from "./actions";
import { middleware } from "./middlewares";

export {
  actions,
  reducer,
  middleware,
};
