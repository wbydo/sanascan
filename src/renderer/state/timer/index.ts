import * as Actions from "./actions";
import middleware from "./middlewares";
import reducer from "./reducers";
import * as types from "./types";

export const timerActions = Actions;
export const timerMiddleware = middleware;
export const timerReducer = reducer;
export const timerActionTypes = types;
