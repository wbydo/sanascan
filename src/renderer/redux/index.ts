import { createStore, applyMiddleware } from "redux";
import { combineReducers} from "redux";

import { composeWithDevTools } from "redux-devtools-extension";

import { reducer as configWindowReducer } from "./state/configWindow";
import { reducer as cursolReducer } from "./state/cursol";
import { reducer as developerModeReducer } from "./state/developerMode";
import { reducer as estimatorReducer } from "./state/estimator";
import { reducer as timerReducer } from "./state/timer";

import { middlewares as timerMiddlewares } from "./cross/timer";
import { middlewares as httpMiddlewares } from "./cross/http";

interface CursolProperty {
  activeColumn: number;
  activeRow: number;
}

export type CursolState = (
  {mode: "normal", direction: "column"} & CursolProperty
) | (
  {mode: "normal", direction: "row"} & CursolProperty
) | (
  {mode: "proposal", direction: "column"} & CursolProperty
);

export interface RootState {
  configWindow: {
    isActive: boolean;
    scanSpeed: number;
  };
  cursol: CursolState;
  developerMode: {
    isActive: boolean;
    estimator: boolean;
    timer: boolean;
  };
  timer: {
    isActive: boolean;
    scanSpeed: number;
  };
  estimator: {
    id: number | null;
    result: string;
  };
}

const middlewares = [
  ...httpMiddlewares,
  ...timerMiddlewares,
];

const enhancer = composeWithDevTools(
  applyMiddleware(...middlewares),
);

export const reducer = combineReducers({
  configWindow: configWindowReducer,
  cursol: cursolReducer,
  developerMode: developerModeReducer,
  estimator: estimatorReducer,
  timer: timerReducer,
});

export const store = createStore(
  reducer,
  enhancer,
);

export { operations } from "./operations";
