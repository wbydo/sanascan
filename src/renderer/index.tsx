import * as React from "react";
import * as ReactDOM from "react-dom";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import createSagaMiddleware from "redux-saga";

import App from "./App";
import reducer from "./reducers";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  applyMiddleware(sagaMiddleware),
);

sagaMiddleware.run(rootSaga);

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root"),
);
