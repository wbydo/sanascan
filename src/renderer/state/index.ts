import { createStore, applyMiddleware } from "redux";
import { combineReducers, Dispatch } from "redux";

import { composeWithDevTools } from "redux-devtools-extension";

import { reducer as configWindowReducer } from "./configWindow";
import { operations as configWindowOperations } from "./configWindow";

import { reducer as cursolReducer } from "./cursol";

import { reducer as estimatorReducer } from "./estimator";
import { middleware as estimatorMiddleware } from "./estimator";
import { operations as estimatorOperations } from "./estimator";

import { reducer as timerReducer } from "./timer";
import { middleware as timerMiddleware } from "./timer";

export interface RootState {
  configWindow: {
    isActive: boolean;
    scanSpeed: number;
  };
  cursol: {
    activeColumn: number;
  };
  timer: {
    id: number | null;
    isActive: boolean;
    scanSpeed: number;
  };
  estimator: {
    id: number | null;
    result: string;
  };
}

const middlewares = [
  timerMiddleware,
  estimatorMiddleware,
];

const enhancer = composeWithDevTools(
  applyMiddleware(...middlewares),
);

export const reducer = combineReducers({
  configWindow: configWindowReducer,
  cursol: cursolReducer,
  estimator: estimatorReducer,
  timer: timerReducer,
});

export const store = createStore(
  reducer,
  enhancer,
);

export const operations = (dispatch: Dispatch) => {
  return {
    configureWindowOpen: configWindowOperations.windowOpen(dispatch),
    resetEstimator: estimatorOperations.reset(dispatch),
    sendKey: estimatorOperations.sendKey(dispatch),
    startFetchEstimatorId: estimatorOperations.fetchId(dispatch),
  };
};
