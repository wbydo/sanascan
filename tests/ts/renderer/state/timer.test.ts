import snapshotDiff from "snapshot-diff";

import { reducer } from "sanascan/renderer/redux";
import { actions } from "sanascan/renderer/redux/state/timer";

import { initialState } from "./initialState";

test("scanSpeedが設定できる", () => {
  const scanSpeed = 234;
  const result = reducer(
    initialState,
    actions.setScanSpeed(scanSpeed),
  );
  expect(
    snapshotDiff( initialState, result),
  ).toMatchSnapshot();
});

describe("isActiveが設定できる", () => {
  for (const i of [true, false]) {
    test( i.toString() + "の場合", () => {
      const result = reducer(
        initialState,
        actions.setActive(i),
      );
      expect(
        snapshotDiff( initialState, result),
      ).toMatchSnapshot();
    });
  }
});
