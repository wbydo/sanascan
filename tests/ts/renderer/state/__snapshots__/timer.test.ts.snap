// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scanSpeedが設定できる 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -13,8 +13,8 @@
    \\"estimator\\": Object {
      \\"id\\": null,
      \\"result\\": \\"\\",
    },
    \\"timer\\": Object {
-     \\"scanSpeed\\": 500,
+     \\"scanSpeed\\": 234,
    },
  }"
`;
